tests/test_reweighting.py

```diff
@@ -0,0 +1,392 @@
+        mean, stddev = annealed_gaussian_def(lam, params)
+        return norm.rvs(loc=mean, scale=stddev, size=(n_samples, 1))
+
+    def logpdf(x, lam, params):
+        mean, stddev = annealed_gaussian_def(lam, params)
+        return np.sum(norm_logpdf(x, loc=mean, scale=stddev))
+
+    def u_fxn(x, lam, params):
+        """unnormalized version of -logpdf"""
+        mean, stddev = annealed_gaussian_def(lam, params)
+        return np.sum(0.5 * ((x - mean) / stddev) ** 2)
+
+    def normalized_u_fxn(x, lam, params):
+        return -logpdf(x, lam, params)
+
+    def reduced_free_energy(lam, params):
+        mean, stddev = annealed_gaussian_def(lam, params)
+        log_z = np.log(stddev * np.sqrt(2 * np.pi))
+        return -log_z
+
+    return u_fxn, normalized_u_fxn, sample, reduced_free_energy
+
+
+def assert_estimator_accurate(estimate_delta_f, analytical_delta_f, ref_params, n_random_trials, atol=5e-3):
+    """for many random parameter sets, assert that the reweighted estimates of
+    delta_f(params) and grad(delta_f)(params) are consistent with analytical_delta_f function"""
+
+    def sample_random_params():
+        mean = ref_params[0] + onp.random.rand()
+        log_sigma = ref_params[1] - onp.random.rand()
+        return np.array([mean, log_sigma])
+
+    f_hat, g_hat = value_and_grad(estimate_delta_f)(ref_params)
+    f_ref, g_ref = value_and_grad(analytical_delta_f)(ref_params)
+
+    onp.testing.assert_allclose(f_hat, f_ref, atol=atol)
+    onp.testing.assert_allclose(g_hat, g_ref, atol=atol)
+
+    for _ in range(n_random_trials):
+        trial_params = sample_random_params()
+        f_hat, g_hat = value_and_grad(estimate_delta_f)(trial_params)
+        f_ref, g_ref = value_and_grad(analytical_delta_f)(trial_params)
+
+        onp.testing.assert_allclose(f_hat, f_ref, atol=atol)
+        onp.testing.assert_allclose(g_hat, g_ref, atol=atol)
+
+
+def test_endpoint_reweighting_1d():
+    """assert that endpoint reweighting estimator for delta_f(params), grad(delta_f)(params) is accurate
+    on tractable 1D system"""
+    onp.random.seed(2022)
+
+    u_fxn, _, sample, reduced_free_energy = make_gaussian_testsystem()
+
+    ref_params = np.zeros(2)
+    ref_delta_f = reduced_free_energy(1.0, ref_params) - reduced_free_energy(0.0, ref_params)
+
+    n_samples = int(1e6)
+
+    samples_0 = sample(0, ref_params, n_samples)
+    samples_1 = sample(1, ref_params, n_samples)
+
+    vec_u = vmap(u_fxn, in_axes=(0, None, None))
+    vec_u_0_fxn = lambda xs, params: vec_u(xs, 0, params)
+    vec_u_1_fxn = lambda xs, params: vec_u(xs, 1, params)
+
+    estimate_delta_f = construct_endpoint_reweighting_estimator(
+        samples_0, samples_1, vec_u_0_fxn, vec_u_1_fxn, ref_params, ref_delta_f
+    )
+    analytical_delta_f = lambda params: reduced_free_energy(1.0, params) - reduced_free_energy(0.0, params)
+
+    assert_estimator_accurate(jit(estimate_delta_f), analytical_delta_f, ref_params, n_random_trials=10, atol=5e-3)
```

Comment:
Out of curiosity, what's the reasoning for `jit`ing the estimator here? (i.e. is to make the test run faster, make the test closer common real-world usage, or something else?)

Reply:
Mostly to make the test run fast!